ui/xui/gl-helpers.cc

```diff
@@ -774,3 +774,20 @@ void SaveScreenshot(GLuint tex, bool flip)
         free(msg);
     }
 }
+
+void export_surface(NV2AState *d, SurfaceBinding *surface)
+{
+     char filename[13];
+     PGRAPHState *pg = &d->pgraph;
+     //uint32_t to support higher bit textures (16, 24, 32). If the xbox does not use these i'll reset to uint8_t/16_t
+     std::vector<uint32_t> png;
+     
+    snprintf(filename, sizeof(filename), "%lu.bmp", surface->vram_addr);
```

Comment:
I'd suggest using `%X` instead as addresses are always printed in hex in xemu (and generally in the industry).